tests/run/cpp_stl_optional.pyx

```diff
@@ -0,0 +1,73 @@
+# ticket: 3293
+# mode: run
+# tag: cpp, werror
+# distutils: extra_compile_args=-std=c++17
```

Comment:
https://github.com/cython/cython/pull/3790 introduces a cpp17 tag (analogous to the cpp11 tag). When that's merged it'd probably be a good idea for this to use the tag rather than setting it through distutils